docs/rfcs/2024-12-28-reports.md

```diff
@@ -0,0 +1,80 @@
+
+## Solution proposal
+
+"Report" feature on the messages that would highlight this message to all group admins and moderators.
+
+Group directory service is also an admin (and will be reduced to moderator in the future), so reported content will be visible to us, so that we can both help group owners to moderate their groups and also to remove the group from directory if necessary.
+
+To the user who have the new version the reports will be sent as a special event, similar to reaction (or it can be simply an extended reaction?) the usual forwarded messages in the same group, but only to moderators (including admins and owners), with additional flag indicating that this is the report.
+
+In the clients with the new version the reports could be shown as a flag, possibly with the counter, on group messages that were reported, in the same line where we show emojis.
+
+If we do that these flags will be seen only by moderators and by the user who submitted the report. When the moderator taps the flag, s/he would see the list of user who reported it, together with the reason.
+
+The downside of the above UX is that it:
+- does not solve the problem of highlighting the problem to admins, particularly if them manage many groups.
+- creates confusion about who can see the reports.
+- further increases data model complexity, as it requires additional table or self-references (as with quotes), as reports can be received prior to the reported content.
+- does not allow admins to see the reported content before it is received by them (would be less important with super-peers).
+
+Alternatively, and it is probably a better option, all reports, both sent by the users and received by moderators across all groups can be shown in the special "chat" called Reports - most likely it should be a separate type of chat that is automatically shown in case there are any reports. The report should be shown as the message sent from a specific member with the group name and and small image visible in the message. The report should allow these actions:
+- moderate the original message,
+- navigate to the original message (requires infinite scrolling, so initially will be only supported on Android and desktop),
+- connect to the user who sent the report - it should be possible even if the group prohibits direct messages. There are two options how this communication can be handled - either by creating a new connection, and shown as normal contacts, or as comments to the report, and sent in the same group connection. The latter approach has the advantage that the interface would not be clutter the interace. The former is much simpler, so should probably be offered as MVP.
+
+This additional chat is necessary, as without it it would be very hard to notice the reports, particularly for the people who moderate multiple groups, and even more so - in our group directory and future super peers.
```

Comment:
another middle option is to show reports indicator on chat list level per conversation for moderators (e.g. some red flag on each group row that has reports, maybe in the same place where unread counter is shown), and inside groups. Inside groups for example like this:
- bar below chat info bar showing "you have x unprocessed reports"
- inside chat info menu itself

the main downside of the first approach seems to be that messages scrolling up would cause reports being missed, that this "you have reports" bar would be visible only to moderators and up seems expected.

this probably would be simpler approach than the latter. another upside is that you instantly see whether you need to react to some important group, or can ignore. also filtering/tags would apply. if it's a single folder without priority you always need to scan all reports

Reply:
I like the idea of having groups with unprocessed reports marked, but it doesn't seem to negate the need for dedicated reports folder...

The scope can easily explode, we need MVP...

Reply:
If you continue on this path you need:
- mark groups with unprocessed reports, inside and outside.
- show reports just for this group (yet another view + API).
- show archived reports for a group.
- engage with report from the group itself.

Given that we are trying to achieve two things:
- give people opportunity to easily submit reports to group moderators (including our directory bot),
- give owners the ability not to miss reports.

Having a single "chat" seems a better compromise, if I have to choose.

The positioning is not a problem - we can always show Reports with unprocessed reports first, and let it bump to its chronological position once all are processed.

I don't even want to have a record for this "chat" and have it virtual instead - returned by backend when the list is loaded only if reports are present.

The main gap in the UI is that we need a view that mixes messages from multiple groups, that is - somehow show both group and member, but it may be the foundation for my dream come true :)

Another question is how and if we show submitted reports to the user who sent it. It probably should be shown somewhere? probably in the conversation itself, but with the marker "only you can see it". Then once we implement comments, the conversation with admin will be under this message. Another option is to show report folder for submitting users as well, but without priority - in which case sent reports will be shown as sent messages (again, they need to show in which group they were sent).

Reply:
I like the chat idea better, I think it has the biggest potential to solve what we are trying to achieve, and I agree that the group by group approach can be very challenging to manage. A few things that aren't clear about the UX and we should probably at least have on the back of our mind:

- As far as I understand it's rare that groups will have a single moderator, I'm guessing at the time of reporting everyone over a certain privilege gets a notification and a new message pushed to this chat, what happens if 1 moderator acts on that report (could be ignore too)? Do we remove the message from the other moderators? Do we show them another moderator already took action? How simple is that?
- On the back of the previous point, what if moderators disagree, shouldn't they be allowed to have a discussion easily?
- We may be focusing too much on the moderators here and too little on the person doing the report, if I do 3 or 4 reports (can be in same or different groups) how would this experience look like without a mechanism similar to "threads" in most chat apps? It isn't clear to me at all how a helpful conversation should be kept and then how we root each message to the right moderators and make the best effort to avoid mistakes done by reporters